app/src/main/java/com/example/crearrepositorio/MainActivity.kt

```diff
@@ -1,11 +1,20 @@
 package com.example.crearrepositorio
 
 import android.os.Bundle
+import androidx.activity.viewModels
 import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
 import com.example.crearrepositorio.databinding.ActivityMainBinding
+import com.example.crearrepositorio.ui.viewModel.MainViewModel
+import com.example.crearrepositorio.ui.viewModel.UiState
+import kotlinx.coroutines.launch
 
 class MainActivity : AppCompatActivity() {
     private lateinit var binding: ActivityMainBinding
+    private val viewModel: MainViewModel by viewModels()
```

Comment:
Buenas! Esto podríamos eliminarlo y usar el `viewModel` que habéis puesto en el `FragmentHome`. 

Creo que podríamos dejar este Activity bastante limpio y no haría falta cargar este `viewModel` acá.